### ISSUE TITLE
Add support for multiple assignees

### ISSUE THREAD

**Issue by normanjoyner** [codetree/feedback#206]
GitHub [just released](https://github.com/blog/2178-multiple-assignees-on-issues-and-pull-requests) support for multiple asignees on issues & PRs. It would be nice to have this ability reflected in Codetree when assigning issues

**Comment by derrickreimer** [comment]
Thanks! Adding this for reference: https://developer.github.com/changes/2016-5-27-multiple-assignees/

**Comment by briandela** [comment]
Unfortunately CodeTree has become a lot less usable for us without this feature. We routinely assign issues to multiple team members - the team member who has time to address an issue first, removes the other person and marks the issue as in progress. 

The codetree `board?assignee=<username>` filter only works for one person assigned to it.

**Comment by kareem** [comment]
@briandela do you have a few moments to chat about this feature?  we're digging into it and would like to better understand how you use it.  If so pls drop me a line at k@codetree.com. Thx

**Comment by kwiens** [comment]
It would be wonderful if you could write up the current functionality and what your plans for the future are.

I'd happily adapt how I assign people issues / pulls if I knew where you were going with it.

**Comment by tom-ld** [comment]
We use multiple assignees here as well, and would like to see this as a feature in Codetree. Frequently our issues require independent work from multiple people (e.g., we may have an issue that needs a change in the backend as well as a matching change in the UI). When that happens, we assign the issue to both the backend and the UI developers.  Now both developers know that they are on the hook for working on the issue. When they filter and sort to see what they're supposed to be working on that day or week, they both see this as being on their plate. They can prioritize and work on the issue separately, but the issue is not resolved until both developers have entered commits identifying their work.

When one developer finishes work, he/she enters a comment marking his/her side of the issue as being addressed (but not resolved) by a commit. When the second developer is done and his/her commit identified, the issue is resolved and assigned to QA to check that the bug has gone away.

This *could* in theory be handled by creating two separate issues, one for each developer. In fact we're kind of forced to do this at the moment. The problem is that we frequently lose track of the dependence between the two separate issues. We end up bouncing two related issues back and forth between developers with comments to capture the "I need you to do this before I can do that" aspect and it gets very confusing.

**Comment by kareem** [comment]
Thanks Tom, that's super helpful to understand how you guys use multiple
assignees.  It's come up a few times recently so this data point is helpful.

In the short term, using Dependencies may be helpful to keep track of
related issues:
http://codetree.helpscoutdocs.com/article/8-dependencies

**Comment by tom-ld** [comment]
Kareem - thanks for the tip on Dependencies, I'll pass that on to our team.

In the mean time I thought I'd pass on a comment made by one of our Dallas developers that might help further (I'm trying to get him to join the Codetree fold :-):

"""
> I believe Github allows multiple people to be allocated to one issue. However, CodeTree does not.
> So far we have been simply kicking an issue back and forth between people by logging a
> comment in CodeTree and changing the assignee. 

I see. I am not using CodeTree and I didn't realized it does not support this. This explains why I don't get a response for some dual-assigned issues.

Kicking things back and forth is an issue for all the features that require UI and API. Most of the time I can work on the UI in parallel with the API and integrate at the end. The kicking back and forth is good for clarifying and questions but it does not work because I don't see the issue anymore and then boom it falls on my plate days before the release. Or it's sitting on my plate and someone else has no idea he is supposed to work on it. 

I don't want to cause too many problems for you guys. How many CodeTree licenses are you paying for now, can I get one?
"""

**Comment by kareem** [comment]
@tom-ld That's super helpful, thanks. Having issues fall through the cracks b/c of the way we currently handle multiple assignees is definitely not great. Will talk this over with the team.

**Comment by tom-ld** [comment]
By the way I did get our Dallas developer to start using Codetree, and now he's active with it all the time. His feedback:

"I just have to say again how much progress CodeTree has made. I am in codetree UI all the time: moving things, commenting, scanning the milestones ahead, checking priorities. Github issues UI creates so much unnecessary stress by mixing everything up. Codetree is very nice."

Take a bow, guys! Good work!

**Comment by kareem** [comment]
Hey @tom-ld thanks so much for the kind words.  Your developer's feedback made our day!  👍

**Comment by kareem** [comment]
Hi all-

We just soft-launched multiple assignee support today.  Please let me know if you see any issues.

Thx,

Kareem
